src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js

```diff
@@ -0,0 +1,218 @@
+import React, { useEffect, useState } from "react";
+import DataTable from "react-data-table-component";
+import PageHeader from "../../../components/Common/PageHeader";
+import { ExportToExcel } from "../../../components/Utilities/Table/ExportToExcel";
+import ReviewCommentMasterModal from "./Validation/ReviewCommentMasterModal";
+import { useDispatch, useSelector } from "react-redux";
+import { getReviewCommentMasterListThunk } from "../../../redux/services/testCases/reviewCommentMaster";
+import { customSearchHandler } from "../../../utils/customFunction";
+import TableLoadingSkelton from "../../../components/custom/loader/TableLoadingSkelton";
+import { Col, Row } from "react-bootstrap";
+
+function ReviewCommentMasterComponent() {
+  const dispatch = useDispatch();
+
+  // // redux state
+  const { reviewCommentMasetrList, isLoading } = useSelector(
+    (state) => state?.reviewCommentMaster
+  );
+
+  const [searchValue, setSearchValue] = useState("");
+  const [filteredRevieCommentMasterList, setFilteredReviewCommentkMasterList] =
+    useState([]);
+
+  const [addEditReviewCommentModal, setAddEditReviewCommentModal] = useState({
+    type: "",
+    data: "",
+    open: false,
+  });
+
+  // Function to handle search button click
+  const handleSearch = () => {
+    const filteredList = customSearchHandler(
+      reviewCommentMasetrList,
+      searchValue
+    );
+    setFilteredReviewCommentkMasterList(filteredList);
+  };
+
+  // Function to handle reset button click
+  const handleReset = () => {
+    setSearchValue("");
+    setFilteredReviewCommentkMasterList(reviewCommentMasetrList);
+  };
+
+  // Update the useEffect to update the filtered list when reviewCommentMasetrList changes
+  useEffect(() => {
```

Comment:
Its standard practice to call useEffect at end , not randomly anywhere.

Reply:
updated the changes